go/libraries/doltcore/sqle/tables.go

```diff
@@ -1720,3 +1722,255 @@ func (t *AlterableDoltTable) constraintNameExists(ctx *sql.Context, name string)
 
 	return false, nil
 }
+
```

Comment:
the rest of the schema change logic is located in `schema/alterschema` we should move this logic there and call to it

Reply:
We should also add some golang integration test for rewriting primary and secondary indexes

Reply:
1. Agree with that refactor
2. Not sure what we're looking for in the integration test. Is it to better assert contents in the index?

Reply:
I'll sync in person